.github/workflows/validation.yml

```
@@ -35,49 +24,106 @@ jobs:
         run: echo "::set-output name=dir::$(yarn config get cacheFolder)"
       - name: Yarn Cache
         uses: actions/cache@v4
-        id: yarn-cache
         with:
-          save-always: true
           path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
           key: yarn-cache-folder-${{ hashFiles('**/yarn.lock', '.yarnrc.yml') }}
           restore-keys: |
             yarn-cache-folder-
+      # We can skip build for all deps because (even for whitelisted) because this is used only to validate yarn.lock and fill cache
```

another nitpicky AI fix because of duplicated "because"
```suggestion
      # We can skip the build for all dependencies, even for those whitelisted, because this process is used only to validate the yarn.lock file and populate the cache.
```